eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreWallet.scala

```diff
@@ -63,7 +64,19 @@ class BitcoinCoreWallet(rpcClient: BitcoinJsonRPCClient)(implicit ec: ExecutionC
 
   def publishTransaction(tx: Transaction)(implicit ec: ExecutionContext): Future[String] = bitcoinClient.publishTransaction(tx)
 
-  def unlockOutpoints(outPoints: Seq[OutPoint])(implicit ec: ExecutionContext): Future[Boolean] = rpcClient.invoke("lockunspent", true, outPoints.toList.map(outPoint => Utxo(outPoint.txid, outPoint.index))) collect { case JBool(result) => result }
+  def unlockOutpoints(outPoints: Seq[OutPoint])(implicit ec: ExecutionContext): Future[Boolean] = {
+    // we unlock utxos one by one and not as a list as it would fail at the first utxo that is not actually lock and the rest would not be processed
+    val futures = outPoints.toList.map(outPoint => Utxo(outPoint.txid, outPoint.index)).map(utxo => {
+      val f = rpcClient.invoke("lockunspent", true, List(utxo))
+      f.recover { case t: Throwable =>
+        logger.warn(s"Cannot unlock utxo=$utxo", t)
+        false
+      }
+      f
+    })
+    val future = Future.sequence(futures).map(_ collect { case JBool(result) => result })
+    future.map(_.forall(b => b))
```

Comment:
I don't know if that's really what we should return. If we failed to unlock a utxo because it was already unlocked, from eclair's point of view it should be a success: the utxo is unlocked which is what we wanted, right?
But maybe the code for that behavior is hard to get right?

Reply:
Yes basically the only case for which we know what happened is when everything worked.

Reply:
There is no way to list utxos and know whether they're locked/unlocked?
Or no way to interpret the error message when trying to unlock and detect the error "can't unlock, it's already unlocked"?

Reply:
I think what's missing here is a method comment explaining what the return value will be.

How about transforming "already unlocked" errors to success and leaving the rest as-is.

Reply:
Ok. Parsing bitcoin RPC error messages is brittle (they may change even between minor releases it it bit us a few times before) but this one seems fairly stable, done